elasticsearch-transport/lib/elasticsearch/transport/transport/http/manticore.rb

```
@@ -43,97 +46,156 @@ module HTTP
+            begin
+              tries += 1
+              yield
+            rescue ::Elasticsearch::Transport::Transport::ServerError => e
+              if @retry_on_status.include?(e.response.status)
+                logger.warn "[#{e.class}] Attempt #{tries} to get response from #{url}" if logger
+                logger.debug "[#{e.class}] Attempt #{tries} to get response from #{url}" if logger
+                if tries <= max_retries
+                  retry
+                else
+                  logger.fatal "[#{e.class}] Cannot get response from #{url} after #{tries} tries" if logger
+                  raise e
+                end
+              else
+                raise e
+              end
+            rescue ::Elasticsearch::Transport::Transport::HostUnreachableError => e
+              logger.error "[#{e.class}] #{e.message} #{e.url}" if logger
 
-                host.delete(:user)     # auth is not supported here.
-                host.delete(:password) # use the headers
+              if @options[:retry_on_failure]
+                logger.warn "[#{e.class}] Attempt #{tries} connecting to #{connection.host.inspect}" if logger
+                if tries <= max_retries
+                  if @options[:reload_on_failure] && pool.alive_urls_count == 0
+                    logger.warn "[#{e.class}] Reloading connections (attempt #{tries} of #{connections.all.size})" if logger
+                    reload_connections!
+                  end
 
-                Connections::Connection.new \
-                  :host => host,
-                  :connection => @manticore
-              },
-              :selector_class => options[:selector_class],
-              :selector => options[:selector]
+                  retry
+                else
+                  logger.fatal "[#{e.class}] Cannot connect to #{connection.host.inspect} after #{tries} tries" if logger
+                  raise e
+                end
+              end
+            rescue Exception => e
```

Can we just `rescue => e` (StandardError)  instead? if the exception actually falls off of StandardError then we should likely let that bubble up